src/main/java/net/dv8tion/jda/internal/entities/ReceivedMessage.java

```diff
@@ -210,13 +210,16 @@ public boolean isPinned()
         return getTextChannel().removeReactionById(getId(), unicode, user);
     }
 
-
     @Nonnull
     @Override
     public ReactionPaginationAction retrieveReactionUsers(@Nonnull Emote emote)
     {
         Checks.notNull(emote, "Emote");
 
+        if (getChannelType() == ChannelType.TEXT) {
+            ((TextChannelImpl) channel).checkPermission(Permission.MESSAGE_HISTORY);
+        }
+
         MessageReaction reaction = this.reactions.stream()
             .filter(r -> r.getReactionEmote().isEmote() && r.getReactionEmote().getEmote().equals(emote))
             .findFirst().orElse(null);
```

Comment:
The docs of [`ReactionPaginationAction#getReaction()`](https://ci.dv8tion.net/job/JDA/javadoc/net/dv8tion/jda/api/requests/restaction/pagination/ReactionPaginationAction.html#getReaction()) suggest that it will always throw if the ReactionPaginationAction is created via `Message#retrieveReactionUsers`.
Do we care enough about having it sometimes not throw (it won't throw if this stream finds something) to not just use
```java
return channel.retrieveReactionUsersById(id, emote);
```
here?

Reply:
I think we should use `channel.retrieveReactionUsersById` regardless since it also does the permission checks. The throw happens if the action is created without a `MessageReaction` instance. If we find one here, it won't throw but the docs say it anyway because users shouldn't rely on this.